EE.C

```diff
@@ -0,0 +1,478 @@
+//   TH2F *bjetpT_elepT_DY = new TH2F("bjetpT_elepT_DY","b-jet pT vs electron pT",100,0,500,100,0,500);
+//   TH2F *bjetpT_mupT_DY = new TH2F("bjetpT_mupT_DY","b-jet pT vs muon pT",100,0,500,100,0,500);
+   
+   Long64_t nentries = fChain->GetEntriesFast();
+
+   float totalN = 122055388;
+   float Lumi = 36.3;
+   float cs = 5765.4;
+   float weight = Lumi * cs/totalN;
+
+   Long64_t nbytes = 0, nb = 0;
+   for (Long64_t jentry=0; jentry<nentries;jentry++) {
+      Long64_t ientry = LoadTree(jentry);
+      if (ientry < 0) break;
+      nb = fChain->GetEntry(jentry);   nbytes += nb;
+      // if (Cut(ientry) < 0) continue;
+
+      run_DY->Fill(run);
+
+      //MET_DY->Fill(met_pt);
+      //dilep_mass_DY->Fill(dil_mass);
+
+      if(lep_pt[0] > 53 &&lep_pt[1] > 53){
+      if(dil_dR > 0.1){
+      if(fabs(lep_dz[0])<0.5 && fabs(lep_dz[1])<0.5){
+
+	  
+	 
+	  //Opposite sign
+	  if(lep_id[0] * lep_id[1] < 0){
+	 
+      //if(dil_chosen != 0) continue;
+	      //particle id
+	      int id1 = lep_id[0];
+	      int id2 = lep_id[1];
+
+	      id_DY->Fill(id1);
+	      id_DY->Fill(id2);
+	      
+	      float Lep_pt[2] = {-999,-999};
+	      float Lep_eta[2] = {-999,-999};
+	      float Lep_phi[2] = {-999,-999};
+	      float Lep_Iso[2] = {-999,-999};
+	      float Lep_E[2] = {-999,-999};
+
+	      float Dil_dPhi[1] = {-999};
+	      float Dil_dR[1] = {-999};
+	      float Dil_mass[1] = {-999};
+	      
+	      float MET[1] = {-999};
+	      
+	      int Njet = 0;
+	      int Nbjet = 0;
+	      int Nnon_bjet = 0;
+	      
+	      vector<float> bJet_pt; bJet_pt.clear();
+	      vector<float> bJet_eta; bJet_eta.clear();
+	      vector<float> bJet_phi; bJet_phi.clear();
+	      vector<float> bJet_E; bJet_E.clear();
+
+	      vector<float> Jet_pt; Jet_pt.clear();
+	      vector<float> Jet_eta; Jet_eta.clear();
+	      vector<float> Jet_phi; Jet_phi.clear();
+	      vector<float> Jet_E; Jet_E.clear();
+	//if(nJets==0)cout<<"Jet: "<<nJets<<endl;
+		  for(int i=0; i<nJets; i++){
+		      if(jet_pt[i] > 30){ //jet pT > 30GeV
+			  if(jet_eta[i] < 5.0){ // jet |eta| <5.0
+
+			      Lep_pt[0] = lep_pt[0];
+
+			      //Number of jet
+			      Njet++;
+
+			      if(jet_btag[i] >= 0.8484) { //b-tagging(WP=0.8484)
+				  Nbjet++;
+				  bJet_pt.push_back(jet_pt[i]);
+				  bJet_eta.push_back(jet_eta[i]);
+				  bJet_phi.push_back(jet_phi[i]);
+				  bJet_E.push_back(jet_E[i]);
+			      }
+			      else Nnon_bjet ++;
+			  }
+		      }
+
+		  }
+	     // }
+
+	      //if(Njet == 0){cout<<"entry : "<<jentry << endl;}
+	     // if(Lep_pt[0] == -999) continue;
+
+
+	      if(Njet < 2) continue;//Number of jet >=2
+
+		 //cout<<"NJet: "<<Njet<<endl; 
+	      njet_DY->Fill(Njet);
+	      nbjet_DY->Fill(Nbjet);
+
+	      njvsnbj_DY->Fill(Nnon_bjet,Nbjet);
+
+	      //if(Nbjet < 1) continue;// >= 1 b-jet
+
+	      for(int i=0; i<nJets; i++){
+		  if(jet_eta[i] == -999.) continue;
```

Comment:
Those are the ntuple contents, as far as I see. Are these fixed size arrays with standard values of -999. for empty entries?
Does nJets not contain the number of actually relevant jets?

Moreover, there seems to be no reason whatsoever to have your second loop segregated from the first, nor do I understand why there's a requirement of jet_phi to be not -999., since jet_eta either catches this in all cases or it's missing from the previous loop.